Documentation updates for v7.0.0 (#348)
auth0/auth0.net: docs-source/index.md

This documentation is supplemental to the official [Auth0 API documentation](https://auth0.com/docs). For more information on the Authentication and Management APIs, you can visit their official documentation by following one of the links below:

> * [Authentication API](https://auth0.com/docs/auth-api)

Reviewer: ```suggestion
* [Authentication API](https://auth0.com/docs/api/authentication)
```

> * [Management API](https://auth0.com/docs/api/v2)

Reviewer: ```suggestion
* [Management API](https://auth0.com/docs/api/management/v2)
```

The rest of this documentation will guide your through the process of installing the various NuGet Packages, as well as some basic examples. You will also find the full references for all the classes and types in the Auth0 .NET SDK.

## Installation

You can install the Auth0 .NET SDK through [NuGet](https://www.nuget.org). Two different NuGet packages are available, one for the Authentication API, and one for the Management API.

### Installing Authentication API SDK

You can install the Authentication API SDK through the [Package Manager Console](https://docs.nuget.org/consume/Package-Manager-Console) inside Visual Studio:

```powershell
Install-Package Auth0.AuthenticationApi
```

Alternatively you can install it through the [Package Manager Dialog](https://docs.nuget.org/consume/package-manager-dialog) by searching for the `Auth0.AuthenticationApi` package.

### Installing Management API SDK
		  
You can install the Management API SDK through the [Package Manager Console](https://docs.nuget.org/consume/Package-Manager-Console) inside Visual Studio:

```powershell
Install-Package Auth0.ManagementApi
```
		  

> Alternatively you can install it through the [Package Manager Dialog](httsp://docs.nuget.org/consume/package-manager-dialog) by searching for the `Auth0.ManagementApi` package.

Reviewer: ```suggestion
Alternatively you can install it through the [Package Manager Dialog](https://docs.nuget.org/consume/package-manager-dialog) by searching for the `Auth0.ManagementApi` package.
```

	.WithResponseType(AuthorizationResponseType.Code)
	.WithClient("abcdef")
	.WithConnection("google-oauth2")

> 	.WithRedirectUrl("https://www.myapp.com/redirect")

Reviewer: 💪

var client = new ManagementApiClient("token", "YOUR_AUTH0_DOMAIN");
```

> You can obtain your Auth0 Domain from the [Application section of the Auth0 Dashboard](https://manage.auth0.com/#/applications) in the settings for your particular application. 

> [!NOTE]

> > For details on how to generate the token, please see the [Access Token for the Management API](https://auth0.com/docs/api/management/v2/tokens)

Reviewer: ```suggestion
> For details on how to generate the token, please see [Access Token for the Management API](https://auth0.com/docs/api/management/v2/tokens)
```

I'd suggest linking to a specific page here, maybe this one?

https://auth0.com/docs/api/management/v2/get-access-tokens-for-test

I'm not sure what the surrounding documentation is here but this is a catching point for a lot of folks new to the Management API. They might think they should generate a token with a really long expiration and just change it out regularly. If it's not there, I'd mention that the manual process is for testing and the CC grant is for production:

https://auth0.com/docs/api/management/v2/get-access-tokens-for-production

Author: I think the page currently being linked to is a good summary and includes both the links you mention on it.

[HttpGet]
public async Task<IActionResult> Get() {
  ...
  var managementConnection = services.Get<IManagementConnection>();
  var management = new ManagementApiClient("token", "YOUR_AUTH0_DOMAIN", managementConnection);
  ...
}
```

`ManagementApiClient` and `AuthenticationApiClient` also implement `IDisposable` and it is good practice to dispose of them when you are done.  

When they have created their own `HttpClient` they will then dispose of it.  Any externally passed `HttpClient` will not be disposed as it is assumed to be shared with other instances.

## Less-common scenarios

> ### Mocking the ManagementApiClient

Reviewer: Nice addition here, very helpful!

> While the `AuthenticationApiClient` has an associated `IAuthenticationApiClient` interface you will find that `ManagementApiClient` does not. The reason for this is that the such an interface would encompass the entire management API surface and that any slight tweak to the Management API - which happens very frequently - would break existing users of that interface.

Reviewer: ```suggestion
While the `AuthenticationApiClient` has an associated `IAuthenticationApiClient` interface you will find that `ManagementApiClient` does not. The reason for this is that the such an interface would encompass the entire management API surface and that any change to the Management API would break existing users of that interface.
```

> Instead, given that the Management API SDK is in fact just a strongly typed wrapper around the REST API, you can instead mock the @Auth0.ManagementApi.IManagementConnection interface which has just three methods:

Reviewer: ```suggestion
Instead, given that the Management API SDK is just a strongly-typed wrapper around the REST API, you can instead mock the @Auth0.ManagementApi.IManagementConnection interface which has just three methods:
```

1. `void SetDefaultHeaders(IDictionary<string, string> headers)` - You can likely ignore this in your mocks
2. `Task<T> GetAsync<T>(Uri uri, ...)` - To handle GET requests
3. `Task<T> SendAsync<T>(HttpMethod method, Uri uri, object body, ...)` - To handle POST/PUT/PATCH/DELETE 

> > TODO: Provide some examples here.

Reviewer: ```suggestion
```

Author: I think the TODO is valid.  I'll put some in after/during the beta but the docs will need to go out before then.
